File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/validation/annotation/DateTimeRangeConstraints.java
package com.jmsoftware.maf.springcloudstarter.validation.annotation;

import com.jmsoftware.maf.springcloudstarter.validation.validator.DateTimeRangeValidator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

/**
 * Description: DateTimeRangeConstraints, change description here.
 *
 * @author 钟俊（zhongjun）, email: zhongjun@toguide.cn, date: 6/3/2021 2:08 PM
 **/
@Documented
@Constraint(validatedBy = DateTimeRangeValidator.class)
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface DateTimeRangeConstraints {
    String message() default "Invalid date time range";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/validation/annotation/DateTimeRangeConstraints.kt
package com.jmsoftware.maf.springcloudstarter.validation.annotation

import com.jmsoftware.maf.springcloudstarter.validation.validator.DateTimeRangeValidator
import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

/**
 * # DateTimeRangeConstraints
 *
 * Description: DateTimeRangeConstraints, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 9:52 AM
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [DateTimeRangeValidator::class])
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class DateTimeRangeConstraints(
    val message: String = "Invalid date time range",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
